djangocms_versioning/models.py

```diff
@@ -71,8 +70,9 @@ def copy(self, created_by):
             content=new_content, created_by=created_by)
         return new_version
 
-    @transition(field=state, source=constants.DRAFT, target=constants.ARCHIVED)
     def archive(self, user):
+        self._set_archive(user)
+        self.save()
         """Change state to ARCHIVED"""
```

Comment:
This docstring has moved from it's rightful place at the beginning of the method! It would generally be good to polish up the docstring/comments on these transition methods - for many devs it won't be obvious why we've split each transition into two methods, so would be good to make some notes in the code about that.